src/providers/bedrock/utils.ts

```diff
@@ -157,47 +157,247 @@ export const transformAI21AdditionalModelRequestFields = (
   return additionalModelRequestFields;
 };
 
-export async function getAssumedRoleCredentials(
+async function assumeRoleWithWebIdentity(token: string, roleArn: string) {
+  const params = new URLSearchParams({
+    Version: '2011-06-15',
+    Action: 'AssumeRoleWithWebIdentity',
+    RoleArn: roleArn,
+    RoleSessionName: `eks-${Date.now()}`,
+    WebIdentityToken: token,
+  });
+
+  const response = await fetch('https://sts.amazonaws.com', {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/x-www-form-urlencoded',
+    },
+    body: params.toString(),
+  });
+
+  if (!response.ok) {
+    const errorMessage = await response.text();
+    console.error({ message: `STS error ${errorMessage}` });
+    throw new Error(`STS request failed: ${response.status}`);
+  }
+
+  const data = await response.text();
+  return parseXml(data);
+}
+
+async function getAssumedWebIdentityCredentials(
   c: Context,
   awsRoleArn: string,
   awsExternalId: string,
-  awsRegion: string,
-  creds?: {
-    accessKeyId: string;
-    secretAccessKey: string;
-    sessionToken?: string;
-  }
+  awsRegion: string
 ) {
-  const cacheKey = `${awsRoleArn}/${awsExternalId}/${awsRegion}`;
   const getFromCacheByKey = c.get('getFromCacheByKey');
   const putInCacheWithValue = c.get('putInCacheWithValue');
 
+  if (env(c).AWS_WEB_IDENTITY_TOKEN_FILE && env(c).AWS_ROLE_ARN) {
+    try {
+      const roleArn = awsRoleArn || env(c).AWS_ROLE_ARN;
+      const cacheKey = `assumed-web-identity-${env(c).AWS_WEB_IDENTITY_TOKEN_FILE}-role-${roleArn}`;
+      const resp = getFromCacheByKey
+        ? await getFromCacheByKey(env(c), cacheKey)
+        : null;
+      if (resp) {
+        return resp;
+      }
+      let token;
+      // for node
+      if (typeof process !== 'undefined' && process.versions?.node) {
```

Comment:
You can directly use the hono getRuntimeKey() function to get the this value.

Reply:
done